### ISSUE TITLE
Errno::ENOENT - No such file or directory @ rb_sysopen - appcenter.podspec.json

### ISSUE THREAD

**Issue by josmithua** [microsoft/appcenter-sdk-react-native#760]
### **Description**

A lot of people are getting the following error when trying to use cocoapods v1.8.4 to install appcenter pods:

`
Errno::ENOENT - No such file or directory @ rb_sysopen - /Users/joshua/pathtomy/Project/ios/Pods/Local Podspecs/appcenter.podspec.json
`

Downgrading to cocoapods v1.7.5 "fixes" the issue.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. use cocoapods v1.8.4, react native v0.61.x, appcenter v2.x.0 
2. `cd ios; pod install`

### **Details**

1. Which version of the App Center SDK are you using?
    - 2.5.0
2. Which OS version did you experience the issue on?
    - iOS
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - N/A
4. What third party libraries are you using?
    - N/A
5. Run the following command and paste the output below: `react-native info`
```
info Fetching system and libraries information...
System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 261.46 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.16.3 - ~/.nvm/versions/node/v10.16.3/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.1, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    Android SDK:
      API Levels: 27, 28
      Build Tools: 28.0.3
      System Images: android-27 | Google Play Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5791312
    Xcode: 11.1/11A1027 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.2 => 0.61.2
```

6. If you're developing for React Native iOS, run the following command and paste the output below: `pod --version`

`1.8.4`

7. Please enable verbose logging for your app using `[MSAppCenter setLogLevel: MSLogLevelVerbose];` before your call to `[AppCenterReactNative register];` (or any other SDK registration). For Android, use `AppCenter.setLogLevel(android.util.Log.VERBOSE);` before your `SoLoader.init` call (you will need to `import com.microsoft.appcenter.AppCenter;`). Include the logs here:

N/A

Related: https://github.com/CocoaPods/CocoaPods/issues/9214

**Comment by annakocheshkova** [comment]
Hey, I see that you've resolved your issue in https://github.com/CocoaPods/CocoaPods/issues/9214, is that right?
Looks like a caching issue or so, but I'll give it a try some time later on a clean project.

**Comment by itzsaga** [comment]
So with CocoaPods 1.7.5 these are installed:
```
Installing AppCenter (2.3.0)
Installing AppCenterReactNativeShared (2.3.0)
Installing appcenter (2.5.0)
Installing appcenter-analytics (2.5.0)
Installing appcenter-crashes (2.5.0)
Installing appcenter-push (2.5.0)
```

With CocoaPods 1.8.4 these are installed:
```
Installing AppCenter (2.5.0)
Installing AppCenterReactNativeShared (2.5.0)
Installing appcenter (2.5.0)
Installing appcenter-analytics (2.5.0)
Installing appcenter-crashes (2.5.0)
Installing appcenter-push (2.5.0)
```

Same `package.json` (see below) yet different results.
```
{
  "name": "AppcenterPushTest",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "appcenter": "2.5.0",
    "appcenter-analytics": "2.5.0",
    "appcenter-crashes": "2.5.0",
    "appcenter-push": "^2.5.0",
    "react": "16.8.1",
    "react-native": "0.61.3"
  },
  "devDependencies": {
    "@babel/core": "^7.5.0",
    "@babel/runtime": "^7.5.0",
    "@react-native-community/eslint-config": "^0.0.3",
    "babel-jest": "^24.1.0",
    "jest": "^24.1.0",
    "metro-react-native-babel-preset": "^0.51.1",
    "react-test-renderer": "16.8.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

**Comment by guperrot** [comment]
AppCenter and AppCenterReactNativeShared should be version 2.5.0 so this is correct with the new version of Cocoapods.
The old version probably is locked because of a Podfile.lock issue, all pods for appcenter npm 2.5.0 should be of version 2.5.0 as well.

**Comment by dhei** [comment]
Maybe try deleting Podfile.lock and `pod install` again?

**Comment by itzsaga** [comment]
Oh, my method for testing is to delete the `Pods` folder and the `Podfile.lock` before running `pod install`. The discrepancy stood out because I did that. Apologies that I missed that piece of info in my earlier post.

**Comment by guperrot** [comment]
I'm still confused, do you have the ENOENT error when all pod versions are correctly set to 2.5.0?

**Comment by itzsaga** [comment]
Yes, I experience the same as @sm1th. In his issue he says that downgrading to `1.7.5` fixed his issue. I was trying to highlight a difference I saw between what is happening with `1.7.5` and `1.8.4` if that's useful. It may not be.

**Comment by guperrot** [comment]
Can you redo the test with Cocoapods `1.7.5` and make sure this time all pod versions in lock file are resolved to 2.5.0? In that scenario, does the ENOENT error occur?

**Comment by itzsaga** [comment]
Just running `pod install` with `1.7.5` nets this (snippet) `Podfile.lock`
```
PODS:
  - appcenter (2.5.0):
    - AppCenterReactNativeShared
    - React
  - appcenter-analytics (2.5.0):
    - AppCenter/Analytics
    - AppCenterReactNativeShared
    - React
  - appcenter-crashes (2.5.0):
    - AppCenter/Crashes
    - AppCenterReactNativeShared
    - React
  - appcenter-push (2.5.0):
    - AppCenter/Push
    - AppCenterReactNativeShared
    - React
  - AppCenter/Analytics (2.3.0):
    - AppCenter/Core
  - AppCenter/Core (2.3.0)
  - AppCenter/Crashes (2.3.0):
    - AppCenter/Core
  - AppCenter/Push (2.3.0):
    - AppCenter/Core
  - AppCenterReactNativeShared (2.3.0):
    - AppCenter/Core (= 2.3.0)

DEPENDENCIES:
  - appcenter (from `../node_modules/appcenter/ios`)
  - appcenter-analytics (from `../node_modules/appcenter-analytics/ios`)
  - appcenter-crashes (from `../node_modules/appcenter-crashes/ios`)
  - appcenter-push (from `../node_modules/appcenter-push/ios`)

SPEC REPOS:
  https://github.com/cocoapods/specs.git:
    - AppCenter
    - AppCenterReactNativeShared

EXTERNAL SOURCES:
  appcenter:
    :path: "../node_modules/appcenter/ios"
  appcenter-analytics:
    :path: "../node_modules/appcenter-analytics/ios"
  appcenter-crashes:
    :path: "../node_modules/appcenter-crashes/ios"
  appcenter-push:
    :path: "../node_modules/appcenter-push/ios"

SPEC CHECKSUMS:
  AppCenter: 9784d2fc998c9bd0b8fbaf4fb9ed69526d12ce1a
  appcenter: bde9923b687332a25607fc1aa9577c9361cfed85
  appcenter-analytics: 0ee7a35def715d4bce58ec435f54161770195166
  appcenter-crashes: 9f9c5647dba19026ff09509576fb7233f69697ff
  appcenter-push: 683b98fcde542864128c828a61751ce8ab39b53d
  AppCenterReactNativeShared: 7ac481cba7f926848a7be76dca4dcb2692df3b06
```

I delete the Pods folder and change `Podfile.lock` to
```
PODS:
  - appcenter (2.5.0):
    - AppCenterReactNativeShared
    - React
  - appcenter-analytics (2.5.0):
    - AppCenter/Analytics
    - AppCenterReactNativeShared
    - React
  - appcenter-crashes (2.5.0):
    - AppCenter/Crashes
    - AppCenterReactNativeShared
    - React
  - appcenter-push (2.5.0):
    - AppCenter/Push
    - AppCenterReactNativeShared
    - React
  - AppCenter/Analytics (2.5.0):
    - AppCenter/Core
  - AppCenter/Core (2.5.0)
  - AppCenter/Crashes (2.5.0):
    - AppCenter/Core
  - AppCenter/Push (2.5.0):
    - AppCenter/Core
  - AppCenterReactNativeShared (2.5.0):
    - AppCenter/Core (= 2.5.0)
```

leaving the rest the same, run `pod install` and get this error:
```
[!] CocoaPods could not find compatible versions for pod "AppCenterReactNativeShared":
  In snapshot (Podfile.lock):
    AppCenterReactNativeShared (= 2.5.0)

  In Podfile:
    appcenter-push (from `../node_modules/appcenter-push/ios`) was resolved to 2.5.0, which depends on
      AppCenterReactNativeShared

It seems like you've changed the constraints of dependency `AppCenterReactNativeShared` inside your development pod `appcenter-push`.
You should run `pod update AppCenterReactNativeShared` to apply changes you've made.
```

Hope this helps.

**Comment by guperrot** [comment]
Are you sure all the npms are pointing to version 2.5.0? Is there a `npm install` discrepancy between package.json and node_modules directory content?

I don't understand why it tries to pull 2.3.0 for some packages but AppCenter cannot mix and match versions of the SDKs without having undefined behavior or build error.

**Comment by itzsaga** [comment]
Doesn't look like it. This is a fresh React Native project. I was testing out the Push functionality. Just looked at the version in `node_modules` and `appcenter`, `appcenter-analytics`, `appcenter-crashes`, `appcenter-link-scripts`, and `appcenter-push` are all `2.5.0`. I've also done a `rm -rf node_modules` to see if reinstalling things fixed it. I agree it seems very odd from my side.

**Comment by annakocheshkova** [comment]
Hey could you also try cleaning both npm and pods cache? (that is `npm cache clean --force` and `rm -rf "${HOME}/Library/Caches/CocoaPods"`)

**Comment by itzsaga** [comment]
Removed `Pods` folder, `Podfile.lock`, followed @annakocheshkova's suggestions and cleaned both the npm and pods cache's, removed `node_modules` folder. With CocoaPods `1.8.4` did `npm install` `cd ios` `pod install` get the error:
`Errno::ENOENT - No such file or directory @ rb_sysopen - /Users/seth/Airship/AppcenterPushTest/ios/Pods/Local Podspecs/appcenter.podspec.json`

**Comment by dhei** [comment]
@itzsaga @sm1th 

With cocoapods v1.8.4, I have tried the following repro steps but I couldn't reproduce the "ENOENT" issue:

1. `react-native init newapp`
2. `npm install appcenter-analytics appcenter-crashes`
3. `cd ios && pod install --repo-update` works without any error. It installed the correct v2.5.0 version without mix-and-match of older version. The cocoapods output is:
```
Detected React Native module pods for appcenter-analytics and appcenter-crashes
Updating local specs repositories
Analyzing dependencies
Downloading dependencies
Installing AppCenter (2.5.0)
Installing AppCenterReactNativeShared (2.5.0)
Installing appcenter-analytics (2.5.0)
Installing appcenter-crashes (2.5.0)
Generating Pods project
Integrating client project
Pod installation complete! There are 30 dependencies from the Podfile and 30 total pods installed.
```
4. Then I tried deleting "ios/Pods" folder and "ios/Podfile.lock" and `pod install` again, still works fine.

My environment information from `react-native info`:
```
info Fetching system and libraries information...
System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 549.26 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.6.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
    Watchman: 4.7.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.1, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    Android SDK:
      API Levels: 23, 24, 25, 26, 27, 28, 29
      Build Tools: 23.0.1, 23.0.2, 23.0.3, 24.0.0, 24.0.1, 24.0.2, 24.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.0, 26.0.1, 26.0.2, 26.0.3, 27.0.3, 28.0.1, 28.0.2, 28.0.3
      System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-25 | Google Play Intel x86 Atom, android-26 | Google Play Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom
      Android NDK: 20.0.5594570
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: 11.1/11A1027 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.3 => 0.61.3 
  npmGlobalPackages:
    create-react-native-app: 0.0.6
    react-native-cli: 2.0.1
    react-native-git-upgrade: 0.2.7
```

**Comment by sm1th** [comment]
@dhei please try installing the appcenter package as well: `npm install appcenter`

**Comment by dhei** [comment]
@sm1th, thanks for the info. I was able to reproduce this error with `npm install appcenter` on cocoapods v1.8.4.

I compared the diff of `pod install` verbose logs of cocoapods v1.7.5 and v1.8.4, but I couldn't find anything. 

We are still investigating this issue, will share more information if we made any progress. In the meanwhile, the workaround is to downgrade cocoapods to v1.7.5.

```ruby
sudo gem uninstall cocoapods
sudo gem install cocoapods -v 1.7.5
```

**Comment by dhei** [comment]
I tried to reproduce this error again today. To my surprise, the same setup as [my previous comment](https://github.com/microsoft/appcenter-sdk-react-native/issues/760#issuecomment-549986831) worked without any errors on cocoapods 1.8.4. See repository https://github.com/dhei/cocoapods-1.8.4-issue.

The steps that **won't** reproduce are:

1. `react-native init newapp`
2. `npm install appcenter-analytics`
3. `cd ios && pod repo update && pod install` (commit https://github.com/dhei/cocoapods-1.8.4-issue/commit/9b73da199315f5061c07b820932524a32376355e)
4. Run the iOS app from Xcode or command line and it worked fine

react-native info output:
```
info Fetching system and libraries information...
System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 106.23 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.6.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
    Watchman: 4.7.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 23, 24, 25, 26, 27, 28, 29
      Build Tools: 23.0.1, 23.0.2, 23.0.3, 24.0.0, 24.0.1, 24.0.2, 24.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.0, 26.0.1, 26.0.2, 26.0.3, 27.0.3, 28.0.1, 28.0.2, 28.0.3
      System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-25 | Google Play Intel x86 Atom, android-26 | Google Play Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom
      Android NDK: 20.0.5594570
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: 11.2/11B52 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.3 => 0.61.3 
  npmGlobalPackages:
    create-react-native-app: 0.0.6
    react-native-cli: 2.0.1
    react-native-git-upgrade: 0.2.7
```

The only things that changed in my environment were Xcode version (11.1 -> 11.2) and iOS SDK version (13.1 -> 13.2). I am not sure if Xcode version would affect this issue. Maybe give it a try?

**Comment by sm1th** [comment]
@dhei after the first pod install, could you try removing the `Pods` folder and the `Podfile.lock` and try the `pod install` again?

**Comment by dhei** [comment]
Thanks @sm1th. 

If I only delete `Pods` folder, `pod install` still works. It is only when I delete `Pods` and `Podfile.lock`, I was able to reproduce the `Errno::ENOENT` error.

1. `react-native init newapp`
2. `npm install appcenter-analytics`
3. `cd ios && pod repo update && pod install` works fine
4. Run the iOS app from Xcode or command line and it worked fine
5. `rm -rf /Pods && pod install` works fine
6. `rm -rf /Pods && rm Podfile.lock && pod install` will fail with error `Errno::ENOENT - No such file or directory @ rb_sysopen - appcenter.podspec.json`

@sm1th Do you check in Podfile.lock? Maybe checking in Podfile.lock as a workaround?

**Comment by th317erd** [comment]
@dhei We have our `Podfile.lock` tracked with source control, and still have the problem. For our team, it was working on my Mac (and then I commited all the changes to source control), and when a team member of mine went to do a `pod install` (after pulling from source control) on his Mac he was having this issue. We compared versions between all our tools, and I had Cocoapods `1.8.3` installed, whereas he had Cocoapods `1.8.4`. I updated to `1.8.4` and retested everything, and I was not able to reproduce the issue on my machine. However, on  his machine, he could checkout `Podfile.lock` and it would break (have this issue), he could remove `Pods` and `Podfile.lock` and it would still have the issue. He also did a `pod repo update`, and that also didn't help. He could remove `node_modules` and `npm i` and he would STILL have the issue. In short, no matter what we "cleaned" he still had this issue. It was only when we did what @sm1th recommended as a workaround that we were able to successfully install the pods with `pod install`. When I get a free moment I will go back through our git history and try to compare the two `Podfile.lock` versions: The tracked version BEFORE the issue, and the tracked version AFTER the workaround. Maybe that will provide a hint to what is going wrong.

**Comment by th317erd** [comment]
So I have more information (which is to say I am just more confused now). I sat down with my coworker (who is still facing this issue) and we ran a few tests. Here are the steps we took:
> 1. First, we cloned our repository fresh, into a new folder, and tried an `npm i && cd ios ; pod install`... and **IT WORKED** with no issues in a fresh clone
> 2. Second, we started to compare things, starting with a file dump `find '.' | sort > ../file_dump.txt` between both clones... we massaged the results for files that don't matter (removing untracked and useless files). We found nothing obvious, and tried `pod install` again same result: `pod install` fails in the original clone, works fine in the fresh clone
> 3. Next we moved the original project clone to a new directory name to rule out any caching issues `mv original_project original_project_random`... `pod install` STILL fails in the original project
> 4. We closed down everything and rebooted the Mac, verifying that `/tmp` was wiped... `pod install` still failing in original project clone
> 5. Now because we found nothing at all (and didn't want to go to the effort to compare file contents) we decided to remove the original project clone, and just use the new one (since it is mysteriously working)
> 6. We remove the old project, and rename the fresh clone to the old name... now `pod install` is failing again :cry: 
> 7. We removed the `Podfile.lock` and tried again: fail.
> 8. We checkout the `Podfile.lock` from git and BOOM! Success!!! :dizzy_face: 

So what did we find out? That we are now more confused then when we started.

The ONLY real difference we can find between my Mac and my coworkers Mac is that I have a case sensitive file system and he doesn't. However, no case issues showed up in our file diffs

**Comment by th317erd** [comment]
@sm1th @dhei the ONLY correlation here I can come up with as a hypothesis is the file sensitivity issue. I can NOT make it fail on my machine, which has a case SENSITIVE file system. Do either of you have a SENSITIVE file system, or are both of you on INSENSITIVE file systems?

This hypothesis also doesn't make a lot of sense to me (I know... I know... why come up with a hypothesis that doesn't make any sense). I could completely understand it the other way around (failing on SENSITIVE file systems but not on INSENSITIVE). Still, I am out of ideas, and this is the only idea I have at the moment. Just figured I would ask the question of you both about file system case sensitivity. :man_shrugging:

**Comment by dhei** [comment]
Hello @th317erd, my macOS file system NOT case-sensitive.

**Comment by sm1th** [comment]
Mine is not case-sensitive as well.
```
   File System Personality:   APFS
   Type (Bundle):             apfs
   Name (User Visible):       APFS
```

**Comment by mhdubose** [comment]
Any updates on this? As it stands, building a react native app for iOS in appcenter appears to be impossible because the hosted runner has CocoaPods 1.8.4. Every build fails with:
```
Errno::ENOENT - No such file or directory @ rb_sysopen - /Users/runner/runners/2.160.1/work/1/s/ios/Pods/Local Podspecs/appcenter.podspec.json
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/specification.rb:646:in `read'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/specification.rb:646:in `checksum'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/lockfile.rb:533:in `block in generate_checksums'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/lockfile.rb:532:in `each'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/lockfile.rb:532:in `generate_checksums'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-core-1.8.4/lib/cocoapods-core/lockfile.rb:428:in `generate'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/installer.rb:744:in `generate_lockfile'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/installer.rb:752:in `write_lockfiles'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/installer.rb:299:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/installer.rb:173:in `integrate'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/installer.rb:162:in `install!'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/command/install.rb:52:in `run'
/usr/local/lib/ruby/gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/lib/cocoapods/command.rb:52:in `run'
/usr/local/lib/ruby/gems/2.6.0/gems/cocoapods-1.8.4/bin/pod:55:in `<top (required)>'
/usr/local/lib/ruby/gems/2.6.0/bin/pod:23:in `load'
/usr/local/lib/ruby/gems/2.6.0/bin/pod:23:in `<main>'
```

**Comment by MatkovIvan** [comment]
Hi @mhdubose ,

As mentioned above, the current workaround is to downgrade CocoaPods. It can be done via [custom build scripts](https://docs.microsoft.com/en-us/appcenter/build/custom/scripts/) on CI

**Comment by mark-plukkido** [comment]
Hi everyone,

I've run my own tests and it seems that if I rename the "appcenter" npm package to something else (mine was "appcenter-react-native"), then pod install works like a charm. Seems like CocoaPods don't like having an AppCenter.podspec and an appcenter.podspec file in the same project.

Unsure about the next steps.

**Comment by lucastonon** [comment]
> Hi everyone,
> 
> I've run my own tests and it seems that if I rename the "appcenter" npm package to something else (mine was "appcenter-react-native"), then pod install works like a charm. Seems like CocoaPods don't like having an AppCenter.podspec and an appcenter.podspec file in the same project.
> 
> Unsure about the next steps.

How do you do that?

**Comment by mark-plukkido** [comment]
Went into node_modules, found the appcenter directory, renamed it, along with changing the name in the package.json of the node_modules/appcenter/package.json and in the node_modules/appcenter/appcenter.podspec. 

Mind you, this will not result in a working state, but does allow `pod install` to successfully complete. So this is not a solution, just a step in the investigation. I'm hoping that the CocoaPods and MSFT folks figure out which way to go about fixing this.

**Comment by guperrot** [comment]
Oh this is an interesting finding @mark-plukkido, maybe we could simply use another podfile name in the npm layer like adding a react-native suffix.

**Comment by Jarred-Sumner** [comment]
This isn't a good fix, but what works for me is:

Copy paste this into`ios/Pods/Local Podspecs/appcenter.podspec.json`:
```json
{
  "name": "appcenter",
  "version": "2.6.1",
  "summary": "Node module that contains common functionality needed to appcenter-* modules",
  "license": "MIT",
  "homepage": "https://github.com/Microsoft/appcenter-sdk-react-native/blob/master/appcenter/README.md",
  "documentation_url": "https://docs.microsoft.com/en-us/appcenter/",
  "authors": {
    "Microsoft": "appcentersdk@microsoft.com"
  },
  "source": {
    "git": "https://github.com/microsoft/appcenter-sdk-react-native.git"
  },
  "source_files": "AppCenterReactNative/**/*.{h,m}",
  "platforms": {
    "ios": "9.0"
  },
  "requires_arc": true,
  "vendored_frameworks": "AppCenterReactNativeShared/AppCenterReactNativeShared.framework",
  "dependencies": {
    "AppCenterReactNativeShared": [],
    "React": []
  }
}
``` 

Replace `2.6.1` with the version that's installed in `node_modules`.

**Comment by tmaly1980** [comment]
So.... what's the actual long-term solution??? This has been going on for 4 months.

**Comment by tmaly1980** [comment]
OK looks like npmjs isn't updated, and because this is a monorepo, I can't reference the appcenter subfolder directly in package.json. I had to fork the monorepo and split the subfolder out into it's own repo:

https://github.com/tmaly1980/appcenter-core

Waiting for someone to update npmjs with the recent merge...

**Comment by annakocheshkova** [comment]
@tmaly1980 We are working on this issue. This should be fixed in one of the next releases.

**Comment by nartc** [comment]
@annakocheshkova Is there a timeline for next release? Our iOS build started failing suddenly this morning as well.

**Comment by dhei** [comment]
> @annakocheshkova Is there a timeline for next release? Our iOS build started failing suddenly this morning as well.

We plan to release SDK this month, hopefully it will be soon. 

Thanks for your patience.

**Comment by SMJ93** [comment]
Hey @annakocheshkova  @dhei is there any update with the release? It's blocking our release :/

**Comment by MatkovIvan** [comment]
@SMJ93 We'll release the SDK before the end of this month

**Comment by saggiyogesh** [comment]
https://github.com/CocoaPods/CocoaPods/issues/9214#issuecomment-548392866

This worked for me.

**Comment by guperrot** [comment]
The new version of the SDK has been released with a fix: [3.0.0](https://github.com/microsoft/appcenter-sdk-react-native/releases/tag/3.0.0).

**Comment by bobber205** [comment]
Confirmed this fixed it for me as well.

**Comment by vahidrn98** [comment]
> 
> 
> The new version of the SDK has been released with a fix: [3.0.0](https://github.com/microsoft/appcenter-sdk-react-native/releases/tag/3.0.0).

i confirm this
